addons/sale/models/sale_order_line.py

```diff
@@ -624,20 +624,21 @@ def _onchange_product_id(self):
             vals['price_unit'] = self.env['account.tax']._fix_tax_included_price_company(self._get_display_price(product), product.taxes_id, self.tax_id, self.company_id)
         self.update(vals)
 
-        title = False
-        message = False
-        result = {}
-        warning = {}
-        if product.sale_line_warn != 'no-message':
-            title = _("Warning for %s", product.name)
-            message = product.sale_line_warn_msg
-            warning['title'] = title
-            warning['message'] = message
-            result = {'warning': warning}
+    @api.onchange('product_id')
+    def _onchange_product_id_warning(self):
+        product = self.product_id
+
+        if product and product.sale_line_warn != 'no-message':
+            result = {
+                'warning': {
+                    'title': _("Warning for %s", product.name),
```

Comment:
`'title': _("Warning for %s", product.display_name),`
it is better to use display_name if you have variant of same template in the SO.

Reply:
Well, the objective of this PR isn't to change existing logic (even less logic more than 5 years old), unless it's buggy/wrong/unclean.
I prefer to leave it like this for now